src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
+import { useState, useCallback } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const findUser = usersFromServer.find(user => user.id === userId);
+
+  return findUser || null;
+}
+
+const todos = todosFromServer.map((todo) => (
+  {
+    ...todo,
+    user: getUser(todo.userId),
+  }
+));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('0');
+  const [newTodos, setNewTodos] = useState(todos);
+  const [isSubmited, setIsSubmitted] = useState(true);
+
+  const createId = useCallback((lastTodos: Todo[]) => {
+    const allIdes = lastTodos.map(todo => todo.id);
+
+    return Math.max(...allIdes) + 1;
+  }, [newTodos]);
```

Comment:
this dependency isn't needed here
```suggestion
  }, []);
```